lib/hawkular/base_client.rb

```diff
@@ -163,6 +163,9 @@ def initialize(message, status_code = 0)
       attr_reader :message, :status_code
     end
 
+    class HawkularConnectionException < HawkularException
```

Comment:
There seems to be the possible exceptions that are throw on unix / windows

http://ruby-doc.org/stdlib-1.9.2/libdoc/socket/rdoc/Socket.html#method-i-connect